### DISCUSSION TITLE
Google Fit export

### DISCUSSION THREAD

**Discussion by brianjmurrell** [OpenTracksApp/OpenTracks#1152] [General]
I realize that direct Google Fit export has been removed from OpenTracks.

So I wonder what, if any, solutions others are using to export data from OpenTracks to Google Fit.

I have tried [FitnessSyncer](www.fitnesssyncer.com) but it doesn't seem to be working.  It says it has synced, but it has not, and It has not even asked me to give it permission to my Google Fit account in  the same way other apps have so I cannot imagine how it could have possibly done any exporting to GF.

**Comment by dennisguse** [comment]
I don't know anything about it; some people use NextCloud.
Can you import files (GPX/KML) into Google Fit?

**Reply by brianjmurrell** [reply]
> some people use NextCloud.

I can see how NextCloud might be a _parking spot_ for the GPX/KML file on it's way to (somehow) getting imported into Google Fit, but I can't see how NextCloud can be the entire solution there.

> Can you import files (GPX/KML) into Google Fit?

You cannot, AFAIU, which is the reason apps such as [FitnessSyncer](https://github.com/OpenTracksApp/OpenTracks/discussions/www.fitnesssyncer.com) and [SyncMyTracks](https://syncmytracks.com/) exist.

Both apps purport to sync data between various health and fitness platforms/apps including Google Fit.  The former doesn't support OpenTracks directly and won't sync a GPX/KML file directly but will allow you to e-mail one to it and it is supposed to sync, but I cannot seem to get it to sync with Google Fit per my original message here.  The latter only syncs between supported platforms and OpenTracks is not one of them.  Neither does it appear to support GPX/KML files.

**Comment by pstorch** [comment]
These synchronization service mostly connect different fitnesstracker cloud services together. As OpenTracks is by itself offline only, this doesn't work.

There is a possibility in FitnessSyncer to read CSV files from cloud drives. 
https://www.fitnesssyncer.com/support/cloud-csv
Maybe this is an option to build. We have CSV export as a feature request already somewhere?

**Comment by brianjmurrell** [comment]
@pstorch Yes.  FitnessSyncer supports more than just CSV.  It also support GPX, KML and others I believe.

The problem with FitnessSyncer is that it is not actually syncing the GPX files that I send to Google Fit, per my original description.

So I was wondering what solutions others might be using that actually work.

**Comment by pstorch** [comment]
@brianjmurrell I guess you have already tried the things mentioned here: https://www.fitnesssyncer.com/support/google-fit
But as I'm not familiar with this tool, I cannot really help.

**Reply by brianjmurrell** [reply]
Yeah, I've tried all of that.  That fact that I never got a request to authorize FitnessSyncer to upload to Google Fit (i.e. from Google) means that nothing is going to work.

I've verified in Google Fit that FitnessSyncer is not one of the **Connected apps**.

Sent them a ticket a day or two ago but have not heard anything back from them.  Although their system also says I have no active tickets, so maybe their ticket system is actually just a black hole.

**Comment by dennisguse** [comment]
@brianjmurrell moving the discussion here again.
What about https://www.goldencheetah.org/ ?

And for real (never used Google Fit), what does it offer that is irresistible to you?
